src/lib/reducer.ts

```diff
@@ -19,62 +19,44 @@ export type Action =
         id: number;
       };
     }
-  | {
-      type: "uncheck";
-      payload: {
-        id: number;
-      };
-    }
   | {
       type: "delete_task";
     };
 
 export const reducer = (state: State, action: Action): State => {
   const task_data = _.cloneDeep(state.task);
-  const comp_data = _.cloneDeep(state.comp);
 
   switch (action.type) {
     case "genelate_task":
-      if (action.payload.data.data !== "") {
-        return {
-          task: [...state.task, action.payload.data],
-          comp: [...state.comp],
-        };
-      } else {
-        return {
-          task: [...state.task],
-          comp: [...state.comp],
-        };
-      }
-
+      return {
+        nextId: state.nextId + 1,
+        task: [
+          ...state.task,
+          { id: state.nextId, data: action.payload.taskName, flag: false },
+        ],
+      };
     case "check":
       for (let i = 0; i < state.task.length; i++) {
         if (state.task[i].id === action.payload.id) {
-          task_data.splice(i, 1);
-          comp_data.push(state.task[i]);
+          task_data[i].flag = !task_data[i].flag;
+
           break;
         }
       }
 
       return {
         task: task_data,
-        comp: comp_data,
+        nextId: state.nextId,
       };
 
-    case "uncheck":
-      for (let i = 0; i < state.comp.length; i++) {
-        if (state.comp[i].id === action.payload.id) {
-          comp_data.splice(i, 1);
-          task_data.push(state.comp[i]);
-          break;
+    case "delete_task":
+      console.log(state.task.length);
+      for (let i = state.task.length - 1; i >= 0; i--) {
+        if (task_data[i].flag === true) {
+          task_data.splice(i, 1);
         }
       }

```

Comment:
`delete_task`はfor文を使わずに、filter関数を用いると、return文だけで完結するよー。

```suggestion
      return {
        task: task_data.filter((t) => t.flag === false),
        nextId: state.nextId,
      };
```